plugins/setting-resources/src/components/MailboxItem.svelte

```diff
@@ -56,24 +46,71 @@
         dangerous: true,
         okLabel: setting.string.Delete,
         action: async () => {
-          getAccountClient()
-            .deleteMailbox(mailbox.mailbox)
-            .then(() => {
-              reloadRequested()
-            })
-            .catch((err: any) => {
-              console.error('Failed to delete mailbox', err)
-            })
+          loadingRequested()
+          try {
+            await deleteMailbox()
+          } catch (err) {
+            console.error('Failed to delete mailbox', err)
+          }
+          reloadRequested()
         }
       },
       undefined
     )
   }
 
-  const openMailboxMenu = (mailbox: MailboxInfo, ev: MouseEvent): void => {
+  async function deleteMailbox (): Promise<void> {
+    await getAccountClient().deleteMailbox(mailbox.mailbox)
+    const client = getClient()
+    const currentUser = getCurrentEmployee()
+    const socialIds = await client.findAll(contact.class.SocialIdentity, {
+      attachedTo: currentUser,
+      type: SocialIdType.EMAIL,
+      value: mailbox.mailbox
+    })
+    for (const socialId of socialIds) {
+      const value = `${socialId.value}#${socialId._id}`
+      await client.updateCollection(
+        socialId._class,
+        socialId.space,
+        socialId._id,
+        socialId.attachedTo,
+        socialId.attachedToClass,
+        socialId.collection,
+        {
+          value,
+          key: buildSocialIdString({ type: SocialIdType.EMAIL, value })
+        }
+      )
+    }
+    const channels = await client.findAll(contact.class.Channel, {
+      attachedTo: currentUser,
+      provider: contact.channelProvider.Email,
+      value: mailbox.mailbox
+    })
+    for (const channel of channels) {
+      await client.removeCollection(
```

Comment:
What do you think about using rate limiter for  `channels ` and `socialIds` requests?

Reply:
There should be only one item in each collection. Loops here are 'just in case'